examples/daal/cpp/source/k_nearest_neighbors/bf_knn_dense_search_batch.cpp

```diff
@@ -0,0 +1,194 @@
+/* file: bf_knn_dense_search_batch.cpp */
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+!  Content:
+!    C++ example of k-Nearest Neighbors search in the batch processing mode.
+!******************************************************************************/
+
+/**
+ * <a name="DAAL-EXAMPLE-CPP-BF_KNN_DENSE_SEARCH_BATCH"></a>
+ * \example bf_knn_dense_search_batch.cpp
+ */
+
+#include "daal.h"
+#include "service.h"
+#include <cstdio>
+
+using namespace std;
+using namespace daal;
+using namespace daal::algorithms;
+using namespace daal::data_management;
+
+/* Input data set parameters */
+const string trainDatasetFileName =
+    "../data/batch/k_nearest_neighbors_train.csv";
+const string testDatasetFileName = "../data/batch/k_nearest_neighbors_test.csv";
+const string groundTruthDistancesFileName =
+    "../data/batch/k_nearest_neighbors_distances_ground_truth.csv";
+const string groundTruthIndicesFileName =
+    "../data/batch/k_nearest_neighbors_indices_ground_truth.csv";
+
+const size_t nFeatures = 5;
+const size_t nNeighbors = 3;
+
+void trainModel(bf_knn_classification::training::ResultPtr &trainingResult);
+void testModel(bf_knn_classification::training::ResultPtr &trainingResult,
+               bf_knn_classification::prediction::ResultPtr &predictionResult);
+void readGroundTruth(NumericTablePtr &groundTruthIndices,
+                     NumericTablePtr &groundTruthDistances);
+void printIndicesResults(
+    NumericTablePtr &testGroundTruth,
+    bf_knn_classification::prediction::ResultPtr &predictionResult);
+void printDistancesResults(
+    NumericTablePtr &testGroundTruth,
+    bf_knn_classification::prediction::ResultPtr &predictionResult);
```

Comment:
The set of functions is inconsistent with other examples - don't introduce unnecessary functions. readGroundTruth is inconsistent. Another inconsistency is printing of results - you should print from main function, if functionality is small enough to don't split into test/train part at all, print from testModel if printing is small enought, or introduce void printResults() for printing.